pkg/template/resource.go

```diff
@@ -92,7 +93,10 @@ func ApplyParamsToResourceTemplate(params []pipelinev1.Param, rt json.RawMessage
 func applyParamToResourceTemplate(param pipelinev1.Param, rt json.RawMessage) json.RawMessage {
 	// Assume the param is valid
 	paramVariable := fmt.Sprintf("$(params.%s)", param.Name)
-	return bytes.Replace(rt, []byte(paramVariable), []byte(param.Value.StringVal), -1)
+	// Escape quotes for backwards compatibility. Needed since we are appending JSON
+	//
```

Comment:
extra comment line?

also is this temporary, if so can we have an issue to track removing it? and if it isn't temporary then maybe "backward compatibility" is a bit vague and we can explain a bit more about what we are being backward compatible with and why?

Reply:
Opened #257 to discuss this.